pkg/runtime/reconciler.go

```diff
@@ -505,7 +513,11 @@ func (r *resourceReconciler) deleteResource(
 		return latest, err
 	}
 	rlog.Info("deleted resource")
-	return latest, nil
+
+	// Return nil resource so that patchStatus attempt is not made.
+	// Underlying K8s resource gets deleted after removing finalizer string and
+	// patchStatus call will return NotFound error.
+	return nil, nil
```

Comment:
Instead of all of the above changes, would it not be easier to just change [this one line](https://github.com/aws-controllers-k8s/runtime/blob/24e0b8ddeaaf7df4bf55ef1a75e4608ea5ec01fb/pkg/runtime/reconciler.go#L647)

from this:

```go
	if ackcompare.IsNotNil(latest) {
```

to this:

```go
	if ackcompare.IsNotNil(latest) && r.rd.IsManaged(latest) {
```

Reply:
that was my 3rd solution on the issue. What do you think about doing this in addition to the current change?
so logically the return from deleteResource also matches the condition change you suggested

Reply:
@surajkota the return from the Delete operation should be the latest observed state of the resource. Not nil.

Reply:
oh right, done!